src/llscan.cc

```diff
@@ -586,69 +590,58 @@ FindJSObjectsVisitor::FindJSObjectsVisitor(SBTarget& target,
 
 
 /* Visit every address, a bit brute force but it works. */
-uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t available) {
-  lldb::SBError error;
-
-  // Test if the map points to a real map.
-  // Try to create an object out of it.
-
-  uint64_t word = target_.GetProcess().ReadUnsignedFromMemory(
-      location, address_byte_size_, error);
-
+uint64_t FindJSObjectsVisitor::Visit(uint64_t location, uint64_t word) {
   v8::Value v8_value(&llv8, word);
 
   v8::Error err;
   // Test if this is SMI
   // Skip inspecting things that look like Smi's, they aren't objects.
   v8::Smi smi(v8_value);
-  if (smi.Check()) {
-    return address_byte_size_;
-  }
+  if (smi.Check()) return address_byte_size_;
 
   v8::HeapObject heap_object(v8_value);
-  if (!heap_object.Check()) {
-    return address_byte_size_;
-  }
-  if (heap_object.IsHoleOrUndefined(err)) {
-    return address_byte_size_;
-  }
-  if (err.Fail()) {
-    return address_byte_size_;
-  }
+  if (!heap_object.Check()) return address_byte_size_;
 
   v8::HeapObject map_object = heap_object.GetMap(err);
-  if (err.Fail() || !map_object.Check()) {
-    return address_byte_size_;
-  }
+  if (err.Fail() || !map_object.Check()) return address_byte_size_;
 
-  if (!IsAHistogramType(heap_object, err)) {
-    return address_byte_size_;
-  }
+  v8::Map map(map_object);
 
-  if (err.Fail()) {
-    return address_byte_size_;
-  }
+  MapCacheEntry map_info;
+  if (map_cache_.count(map.raw()) == 0) {
+    v8::Value::InspectOptions inspect_options;
+    inspect_options.detailed = false;
+    inspect_options.print_map = false;
+    inspect_options.print_source = false;
+    inspect_options.string_length = 0;
```

Comment:
There's nothing wrong with this change except that it reminds me I ought to remove the inspect_options parameter from GetTypeName. It shouldn't need them as it only ever returns a type name (not the source or the map and it doesn't truncate it if it's a long string).